### ISSUE TITLE
False positive in generic.nginx.security.missing-internal.missing-internal

### ISSUE THREAD

**Issue by AlexanderSilaev** [returntocorp/sgrep-rules#3057]
**Describe the bug**
The patterns for `missing-internal` match the dollar sign character in the lines following the `proxy_pass` line.

**To Reproduce**
```
location / {
    # ok: missing-internal
    proxy_pass  http:/backend:42/;
    set         $false  'positive';
}
```

**Expected behavior**
Test passes.

**Priority**
How important is this to you?
- [ ] P0: blocking me from making progress
- [x] P1: this will block me in the near future
- [ ] P2: annoying but not blocking me

**Comment by p4p3r** [comment]
Hello @AlexanderSilaev, thanks for reporting this!
It should be addressed by https://github.com/returntocorp/semgrep-rules/pull/3101.
